File: js/bugSpecies.js

function nameGen(){
	var nm1 = ["Abnormal","Abrasive","Absorbtion","Abstract","Acid","Acidic","Acoustic","Actor","Adapting","Aggressive","Anchor","Ancient","Angel","Angelic","Arctic","Arid","Aromatic","Assault","Attack","Audience","Bamboo","Barren","Basin","Basking","Battle","Bell","Berserker","Betrayal","Biting","Bitter","Black","Blade","Blind","Blood","Blossom","Blue","Bold","Bomber","Bone","Bony","Boulder","Brain","Brass","Bright","Brilliant","Bronze","Bubble","Bulb","Burrowing","Bush","Butter","Button","Cacti","Candid","Cannon","Cemetery","Champion","Chaos","Chimera","Cloud","Coast","Colossal","Community","Conquerer","Corrupting","Corruption","Creeping","Crimson","Crooked","Crown","Crypt","Curved","Dancing","Dark","Darling","Death","Deceit","Defender","Defiant","Defiler","Delivery","Demonic","Diamond","Diligent","Disguise","Draconian","Dragon","Dread","Dream","Drone","Drunk","Dueling","Ebon","Echo","Electric","Emerald","Emigrating","Emigration","Enchanted","Enchanting","Entangling","Ethereal","Evasive","Exalted","Exploring","Faded","Faint","Fake","False","Familiar","Fancy","Fire","Flamboyant","Flower","Fog","Forging","Fragrant","Frigid","Furry","Fuzzy","Garden","Gardening","Gatherer","Ghost","Giant","Gilded","Glass","Gliding","Gold","Golden","Grand","Grave","Great","Green","Grim","Grooming","Guardian","Gummy","Hairy","Hammer","Harvest","Harvester","Hill","Hissing","Hollow","Humming","Hypnotic","Ice","Infernal","Iron","Ivory","Jade","Jewel","Jolly","Juvenile","Lavish","Leather","Light","Lumber","Luminous","Macabre","Majestic","Mammoth","Marble","Marked","Masked","Milk","Minor","Monster","Monstrous","Mud","Mushy","Night","Nimble","Nocturnal","Noxious","Nursing","Oceanic","Ornate","Perfumed","Phoenix","Plump","Primal","Prime","Putrid","Radiant","Rainbow","Red","Regal","Rigid","Ring","Ringed","River","Rock","Royal","Rubble","Ruby","Ruin","Sanguine","Sapphire","Scaly","Scented","Servant","Shadow","Silk","Silky","Skeletal","Skeleton","Skinny","Smiling","Smoke","Sorrow","Spiked","Spiky","Spotted","Stone","Storm","Striped","Sugar","Thicket","Throne","Tomb","Tower","Tree","Tube","Umbrella","Vagabond","Velvet","Vibrant","Volatile","Volcano","Wandering","Wax","White","Wicked"];
	var nm2 = ["Amphipod","Ant","Aphid","Arachnid","Bee","Beetle","Billbug","Borer","Bug","Bumblebee","Butterfly","Caterpillar","Centipede","Chigger","Cicada","Cockroach","Crawler","Cricket","Damselfly","Dragonfly","Earwig","Flatworm","Flea","Fly","Grasshopper","Grub","Hopper","Horntail","Katydid","Lacewing","Ladybird","Larva","Leech","Locust","Longhorn","Louse","Maggot","Mantid","Mantis","Mayfly","Millipede","Mite","Mosquito","Moth","Nepidae","Parasite","Peripatus","Pseudoscorpion","Psocid","Pupa","Scarab","Silverfish","Slater","Slug","Snail","Sowbug","Spider","Spittlebug","Springtail","Stick Insect","Stonefly","Termite","Thrip","Tick","Wasp","Weevil","Weta","Worm"];

	var br = "";

	$('#placeholder').css('textTransform', 'capitalize');
	var element = document.createElement("div");
	element.setAttribute("id", "result");
	
	for(i = 0; i < 10; i++){
		rnd2 = Math.floor(Math.random() * nm2.length);
		rnd = Math.floor(Math.random() * nm1.length);
		names = nm1[rnd] + " " + nm2[rnd2];
		nm1.splice(rnd, 1);
		nm2.splice(rnd2, 1);
		br = document.createElement('br');	
		element.appendChild(document.createTextNode(names));
		element.appendChild(br);
	}
	if(document.getElementById("result")){
		document.getElementById("placeholder").removeChild(document.getElementById("result"));
	}		
	document.getElementById("placeholder").appendChild(element);
}

File: generators/bugSpecies.py
__all__ = ['bugSpecies']

# Don't look below, you will not understand this Python code :) I don't.

from js2py.pyjs import *
# setting scope
var = Scope( JS_BUILTINS )
set_global_object(var)

# Code follows:
var.registers(['nameGen'])
@Js
def PyJsHoisted_nameGen_(this, arguments, var=var):
    var = Scope({'arguments':arguments, 'this':this}, var)
    var.registers(['nm1', 'nm2', 'result'])
    var.put('nm1', Js([Js('Abnormal'), Js('Abrasive'), Js('Absorbtion'), Js('Abstract'), Js('Acid'), Js('Acidic'), Js('Acoustic'), Js('Actor'), Js('Adapting'), Js('Aggressive'), Js('Anchor'), Js('Ancient'), Js('Angel'), Js('Angelic'), Js('Arctic'), Js('Arid'), Js('Aromatic'), Js('Assault'), Js('Attack'), Js('Audience'), Js('Bamboo'), Js('Barren'), Js('Basin'), Js('Basking'), Js('Battle'), Js('Bell'), Js('Berserker'), Js('Betrayal'), Js('Biting'), Js('Bitter'), Js('Black'), Js('Blade'), Js('Blind'), Js('Blood'), Js('Blossom'), Js('Blue'), Js('Bold'), Js('Bomber'), Js('Bone'), Js('Bony'), Js('Boulder'), Js('Brain'), Js('Brass'), Js('Bright'), Js('Brilliant'), Js('Bronze'), Js('Bubble'), Js('Bulb'), Js('Burrowing'), Js('Bush'), Js('Butter'), Js('Button'), Js('Cacti'), Js('Candid'), Js('Cannon'), Js('Cemetery'), Js('Champion'), Js('Chaos'), Js('Chimera'), Js('Cloud'), Js('Coast'), Js('Colossal'), Js('Community'), Js('Conquerer'), Js('Corrupting'), Js('Corruption'), Js('Creeping'), Js('Crimson'), Js('Crooked'), Js('Crown'), Js('Crypt'), Js('Curved'), Js('Dancing'), Js('Dark'), Js('Darling'), Js('Death'), Js('Deceit'), Js('Defender'), Js('Defiant'), Js('Defiler'), Js('Delivery'), Js('Demonic'), Js('Diamond'), Js('Diligent'), Js('Disguise'), Js('Draconian'), Js('Dragon'), Js('Dread'), Js('Dream'), Js('Drone'), Js('Drunk'), Js('Dueling'), Js('Ebon'), Js('Echo'), Js('Electric'), Js('Emerald'), Js('Emigrating'), Js('Emigration'), Js('Enchanted'), Js('Enchanting'), Js('Entangling'), Js('Ethereal'), Js('Evasive'), Js('Exalted'), Js('Exploring'), Js('Faded'), Js('Faint'), Js('Fake'), Js('False'), Js('Familiar'), Js('Fancy'), Js('Fire'), Js('Flamboyant'), Js('Flower'), Js('Fog'), Js('Forging'), Js('Fragrant'), Js('Frigid'), Js('Furry'), Js('Fuzzy'), Js('Garden'), Js('Gardening'), Js('Gatherer'), Js('Ghost'), Js('Giant'), Js('Gilded'), Js('Glass'), Js('Gliding'), Js('Gold'), Js('Golden'), Js('Grand'), Js('Grave'), Js('Great'), Js('Green'), Js('Grim'), Js('Grooming'), Js('Guardian'), Js('Gummy'), Js('Hairy'), Js('Hammer'), Js('Harvest'), Js('Harvester'), Js('Hill'), Js('Hissing'), Js('Hollow'), Js('Humming'), Js('Hypnotic'), Js('Ice'), Js('Infernal'), Js('Iron'), Js('Ivory'), Js('Jade'), Js('Jewel'), Js('Jolly'), Js('Juvenile'), Js('Lavish'), Js('Leather'), Js('Light'), Js('Lumber'), Js('Luminous'), Js('Macabre'), Js('Majestic'), Js('Mammoth'), Js('Marble'), Js('Marked'), Js('Masked'), Js('Milk'), Js('Minor'), Js('Monster'), Js('Monstrous'), Js('Mud'), Js('Mushy'), Js('Night'), Js('Nimble'), Js('Nocturnal'), Js('Noxious'), Js('Nursing'), Js('Oceanic'), Js('Ornate'), Js('Perfumed'), Js('Phoenix'), Js('Plump'), Js('Primal'), Js('Prime'), Js('Putrid'), Js('Radiant'), Js('Rainbow'), Js('Red'), Js('Regal'), Js('Rigid'), Js('Ring'), Js('Ringed'), Js('River'), Js('Rock'), Js('Royal'), Js('Rubble'), Js('Ruby'), Js('Ruin'), Js('Sanguine'), Js('Sapphire'), Js('Scaly'), Js('Scented'), Js('Servant'), Js('Shadow'), Js('Silk'), Js('Silky'), Js('Skeletal'), Js('Skeleton'), Js('Skinny'), Js('Smiling'), Js('Smoke'), Js('Sorrow'), Js('Spiked'), Js('Spiky'), Js('Spotted'), Js('Stone'), Js('Storm'), Js('Striped'), Js('Sugar'), Js('Thicket'), Js('Throne'), Js('Tomb'), Js('Tower'), Js('Tree'), Js('Tube'), Js('Umbrella'), Js('Vagabond'), Js('Velvet'), Js('Vibrant'), Js('Volatile'), Js('Volcano'), Js('Wandering'), Js('Wax'), Js('White'), Js('Wicked')]))
    var.put('nm2', Js([Js('Amphipod'), Js('Ant'), Js('Aphid'), Js('Arachnid'), Js('Bee'), Js('Beetle'), Js('Billbug'), Js('Borer'), Js('Bug'), Js('Bumblebee'), Js('Butterfly'), Js('Caterpillar'), Js('Centipede'), Js('Chigger'), Js('Cicada'), Js('Cockroach'), Js('Crawler'), Js('Cricket'), Js('Damselfly'), Js('Dragonfly'), Js('Earwig'), Js('Flatworm'), Js('Flea'), Js('Fly'), Js('Grasshopper'), Js('Grub'), Js('Hopper'), Js('Horntail'), Js('Katydid'), Js('Lacewing'), Js('Ladybird'), Js('Larva'), Js('Leech'), Js('Locust'), Js('Longhorn'), Js('Louse'), Js('Maggot'), Js('Mantid'), Js('Mantis'), Js('Mayfly'), Js('Millipede'), Js('Mite'), Js('Mosquito'), Js('Moth'), Js('Nepidae'), Js('Parasite'), Js('Peripatus'), Js('Pseudoscorpion'), Js('Psocid'), Js('Pupa'), Js('Scarab'), Js('Silverfish'), Js('Slater'), Js('Slug'), Js('Snail'), Js('Sowbug'), Js('Spider'), Js('Spittlebug'), Js('Springtail'), Js('Stick Insect'), Js('Stonefly'), Js('Termite'), Js('Thrip'), Js('Tick'), Js('Wasp'), Js('Weevil'), Js('Weta'), Js('Worm')]))
    var.put('result', Js([]))
    #for JS loop
    var.put('i', Js(0.0))
    while (var.get('i')<Js(10.0)):
        try:
            var.put('rnd2', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm2').get('length'))))
            var.put('rnd', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm1').get('length'))))
            var.put('names', ((var.get('nm1').get(var.get('rnd'))+Js(' '))+var.get('nm2').get(var.get('rnd2'))))
            var.get('nm1').callprop('splice', var.get('rnd'), Js(1.0))
            var.get('nm2').callprop('splice', var.get('rnd2'), Js(1.0))
            var.get('result').callprop('push', var.get('names'))
        finally:
                (var.put('i',Js(var.get('i').to_number())+Js(1))-Js(1))
    pass
    return var.get('result')
PyJsHoisted_nameGen_.func_name = 'nameGen'
var.put('nameGen', PyJsHoisted_nameGen_)
pass
pass


# Add lib to the module scope
bugSpecies = var.to_python()